frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -780,6 +782,12 @@ void Visitor::checkNoReceiverClauseOnPrimaryMethod(const Function* node) {
   }
 }
 
+void Visitor::checkLambdaDeprecated(const Function* node) {
+  if (node->kind() != Function::LAMBDA) return;
+  warn(node, "'lambda' syntax is deprecated, please construct anonymous "
+             "procedures using the 'proc' keyword instead");
+}
+
```

Comment:
Can't you use the `Deprecation` error type here?

Reply:
Right now that requires information that can't be computed for this such as target IDs. I'd have to go through and refactor it. In general we probably need a slightly different strategy for deprecating language features.